plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1088,6 +1103,125 @@ def run(
             f"{self.fract_deflected*100}%"
         )
 
+        # Simulation has not run, because creating new particles changes the simulation
+        self._has_run = True
+
+        return self.results_dict
+
+    @property
+    def results_dict(self):
+        r"""
+        A dictionary containing the results of the simulation.
+
+        'source' : `~numpy.ndarray`
+            The source location vector, in meters.
+
+        'detector' : `~numpy.ndarray`
+            The detector location vector in meters.
+
+        'mag' : float
+            The system magnification.
+
+        'nparticles' : int
+            Number of particles in the simulation.
+
+        'max_deflection' : `~astropy.units.Quantity`
+            The maximum deflection experienced by a particle
+            in the simulation, in radians.
+
+        'xloc' :  `~numpy.ndarray`, [nparticles,]
+            The x-coordinate location where each particle hit the
+            detector plane, in meters.
+
+        'yloc' :  `~numpy.ndarray`, [nparticles,]
+            The y-coordinate location where each particle hit the
+            detector plane, in meters.
+
+        'v' : `~numpy.ndarray`, [nparticles, 3]
+            The velocity of each particle when it hit the detector plane,
+            in meters per second. The velocity is in a coordinate system
+            relative to the detector plane. The components are
+            [normal, horizontal, vertical] relative to the detector plane
+            coordinates.
+
+        'xloc0' :  `~numpy.ndarray`, [nparticles,]
+            The x-coordinate location where each particle would have hit the
+            detector plane if the grid fields were zero, in meters. Useful
+            for calculating the source profile.
+
+        'yloc0' :  `~numpy.ndarray`, [nparticles,]
+            The y-coordinate location where each particle would have hit the
+            detector plane if the grid fields were zero, in meters. Useful
+            for calculating the source profile.
+
+        'v0' : `~numpy.ndarray`, [nparticles, 3]
+            The velocity of each particle when it hit the detector plan if the
+            grid fields were zero, in meters per second. The velocity is in a
+            coordinate system relative to the detector plane. The components are
+            [normal, horizontal, vertical] relative to the detector plane
+            coordinates.
```

Comment:
This was not very readable/clear in RTD build...

![image](https://user-images.githubusercontent.com/29869348/131402712-18ba73b7-b7e3-4c46-9605-1d23b942feb0.png)


So, I think it would look better as a table...

![image](https://user-images.githubusercontent.com/29869348/131402798-2f45cde5-559e-4046-9428-81545d53d78d.png)


```suggestion
        .. list-table:: Dictionary keys and descriptions.
           :width: 100%
           :widths: 1 1 4
           :header-rows: 1

           * - Key
             - Type
             - Description
           * - ``"source"``
             - `~numpy.ndarray`
             - The source location vector, in meters.
           * - ``"detector"``
             - `~numpy.ndarray`
             - The detector location vector, in meters.
           * - ``"mag"``
             - `float`
             - The system magnification.
           * - ``"nparticles"``
             - `int`
             - Number of particles in the simulation.
           * - ``"max_deflection"``
             - `~astropy.units.Quantity`
             - The maximum deflection experienced by a particle in the
               simulation, in radians.
           * - ``"xloc"``
             - `~numpy.ndarray`, [``nparticles``,]
             - The x-coordinate location where each particle hit the
               detector plane, in meters.
           * - ``"yloc"``
             - `~numpy.ndarray`, [``nparticles``,]
             - The y-coordinate location where each particle hit the
               detector plane, in meters.
           * - ``"v"``
             - `~numpy.ndarray`, [``nparticles``, 3]
             - The velocity of each particle when it hits the detector
               plane, in meters per second. The velocity is in a
               coordinate system relative to the detector plane. The
               components are [normal, horizontal, vertical] relative
               to the detector plane coordinates.
           * - ``"xloc0"``
             - `~numpy.ndarray`, [``nparticles``,]
             - The x-coordinate location where each particle would have
               hit the detector plane if the grid fields were zero, in
               meters. Useful for calculating the source profile.
           * - ``"yloc0"``
             - `~numpy.ndarray`, [``nparticles``,]
             - The y-coordinate location where each particle would have
               hit the detector plane if the grid fields were zero, in
               meters. Useful for calculating the source profile.
           * - ``"v0"``
             - `~numpy.ndarray`, [``nparticles``, 3]
             - The velocity of each particle when it hit the detector
               plan if the grid fields were zero, in meters per second.
               The velocity is in a coordinate system relative to the
               detector plane. The components are [normal, horizontal,
               vertical] relative to the detector plane coordinates.
```

Reply:
Perfect, thanks, I tried to format this a few ways but kept getting errors.